test/packages/with-kind/kubernetes/_dev/deploy/k8s/kustomization.yaml

```
@@ -0,0 +1,10 @@
+resources:
+  - pv.yaml
+  - pvc.yaml
+  - cronjob.yaml
+  - job.yaml
+  - daemonset.yaml
+  - configmap.yaml
+  - rq.yaml
+  - ss.yaml
+  - github.com/kubernetes/kube-state-metrics?ref=main
```

Please, use a reference to a released version here to avoid unpredictable builds.
```suggestion
  - github.com/kubernetes/kube-state-metrics?ref=abe3fd3184e16893b5a47196f90a94ed13e1b04d
```
Or a tag.
```suggestion
  - github.com/kubernetes/kube-state-metrics?ref=v2.7.0
```

---

The idea of using `main` was to make it unnecessary to update KSM version every time. `main` should always be pointing to the latest version.

---

This is a test in `elastic-package` that happens to reference KSM repository, we are not testing the functionality of the kubernetes integration or its integration with KSM here.
Having this reference to main can break elastic-package builds unexpectedly, when it doesn't have any relationship to KSM.

There is no advantage on using main here, and can be a source of problems.